From 93ec31cfce6c574e834951a3aa15b9b1b26c3b3d Mon Sep 17 00:00:00 2001
From: varun <varunx.jeevangoudar@intel.com>
Date: Mon, 4 Mar 2019 12:27:47 +0530
Subject: [PATCH] Fix the behavior of keyguard bouncer in a corner case

With Android Automotive, there's a cancel button on the bouncer
that allows return to UserSwitcher. Due to specific implementation of
CarStatusBarKeyguardViewManager, all KeyguardViews will be destroyed.
With below steps, user can input PIN the 6th time. But it won't unlock
the device even if it's the correct PIN.

Test: 1. Set screen lock with pin, password or pattern
Test:    in Settings ->Security & location
Test: 2. Lock the device with Power key
Test: 3. Input wrong pin/password for 5 times, 30s timeout is prompted
Test: 4. Quickly click cancel button and select the same user
Test: 5. User can input PIN/password again

Change-Id: Ibe6162b0708ea41989421c4ec6975928b7c12fb0
Tracked-On: OAM-76657
Signed-off-by: varun <varunx.jeevangoudar@intel.com>
---
 .../src/com/android/systemui/statusbar/phone/KeyguardBouncer.java        | 1 +
 1 file changed, 1 insertion(+)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardBouncer.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardBouncer.java
index c74d09d..f7e6cc0 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardBouncer.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardBouncer.java
@@ -226,6 +226,7 @@ public class KeyguardBouncer {
             mShowingSoon = false;
             if (mExpansion == EXPANSION_VISIBLE) {
                 mKeyguardView.onResume();
+                mKeyguardView.resetSecurityContainer();
             }
             StatsLog.write(StatsLog.KEYGUARD_BOUNCER_STATE_CHANGED,
                 StatsLog.KEYGUARD_BOUNCER_STATE_CHANGED__STATE__SHOWN);
-- 
1.9.1

